doc/memos/rdm-draft-alamos-ieee802154-radio-hal.md

```diff
@@ -498,267 +430,848 @@ Note that the Radio Ops interface only implements a few get functions. The
+using bit flags. The Radio HAL header defines a set of `ieee802154_radio_has_*`
+functions that check whether the radio supports a capability.
+See Appendix A for a full list of capabilities.
+
+A Radio HAL implementation MUST indicate all capabilities supported by the
+device and driver implementation.
+
+# 7 Future Proof Considerations
+
+The Radio HAL is designed to be agnostic to different versions of the
+IEEE802.15.4 standard.  A single radio device typically implements hardware
+acceleration for only one standard, whereas different standards are not always
+compatible. As an example, IEEE802.15.4--2006 devices do not support Enhanced
+Acknowledgement packets which are required by the TSCH layer of
+IEEE802.15.4--2012.  For compatibility, a software MAC can provide such
+functionality. The Radio HAL adapts considerations to enable different versions
+of the IEEE802.15.4 MAC on top of the abstraction layer.
+
+### Transmission Modes
+
+The Radio HAL interface defines three transmission modes to allow sending
+frames using (i) CSMA-CA, (ii) CCA, or (iii) directly without any contention
+mechanism.
+In that way, a MAC layer can easily send data frames benefiting from hardware
+accelerated CSMA-CA or Beacons that have to meet timing constraints and thus,
+require a direct send function.
+
+A HAL implementation can provide several transmit modes, but it MUST implement
+at least one. It is recommended that the implementation provides modes that
+exploit the internal devices capabilities. Implementing a direct mode is
+desired for software MAC layers on top.
+
+### PHY Definition
+
+PHY definitions are specific to a IEEE802.15.4 version. As an example, older
+standards define PHY channels with a `channel number`. In modern standards,
+channels are represented using a (`channel number`, `channel page`, `channel
+modulation`) tuple. The `config_phy` function is called with a pointer to a
+`ieee802154_phy_conf_t` structure which describes the PHY configuration. In
+order to support newer versions, this structure can be extended without changing
+the Radio HAL API.
+
+### Future Radio Operations
+
+The Radio Operations interface `radio_ops` can be extended to support
+functionalities of newer standards. As an example, most SubGHz radios support a
+Listen Before Talk feature that can be implemented as a new and optional
+operation.
+
+# 8 Acknowledgements
+
+Thanks to Peter Kietzmann, Benjamin Valentin, Marian Buschsieweke, Leandro
+Lanzieri and Martine Lenders for their reviews, comments and suggestions.
+
+# 9 References
+
+- [Uniform Network Stack Integration](https://github.com/RIOT-OS/RIOT/issues/13771)
+- [Top and Bottom Halves](http://www.makelinux.net/ldd3/chp-10-sect-4.shtml)
+
+# 10 Revision
+
+- Rev6: Relax Abstract State Machine transitions
+- Rev5: Add Source Address Matching
+- Rev4: Add Async behavior
+- Rev3: Merge Device Driver and HAL
+- Rev2: Update sleep preconditions
+- Rev1: Address on/off functions 
+- Rev0: initial document
```

Comment:
I think these can go, as the doc was not part of master.

Reply:
alright